interfaces/repo.go

```diff
@@ -747,35 +747,30 @@ func (r *Repository) RemoveSnap(snapName string) error {
 
 // DisconnectSnap disconnects all the connections to and from a given snap.
 //
-// The return value is a list of names that were affected.
-func (r *Repository) DisconnectSnap(snapName string) ([]string, error) {
+// The return value is a set of snaps that were affected.
+func (r *Repository) DisconnectSnap(snapName string) (map[*snap.Info]bool, error) {
```

Comment:
So the returned map of affected snaps is further filled in ```setupProfilesForSnap``` in handlers.go when calling  ```m.repo.ConnectedSnaps```. Then ```setupAffectedSnaps``` uses that map - but you are correct it only uses the name from ```snap.Info```

The basic scenario I'm trying to address is:

Scenario 1 - which is what this PR addresses
* snap A is being installed for the first time and has plug that auto-connects to snap B slot
  * DisconnectSnap returns an empty map as there was nothing to disconnect
  * snap A plug is auto-connected to snap B slot
  * the empty map of affected snaps is passed to m.repo.ConnectedSnaps, which fills it with snapB info
  * map now contains one affected snap info (snap B's)
  * setupAffectedSnaps sets up security for snap B

Scenario 2
* A new revision of snap A is being installed which has plug connected to snap B slot
  * DisconnectSnap returns a map with one entry (snapB info)
  * snap A plug is auto-connected to snap B slot
  * the current map of affected snaps is passed to m.repo.ConnectedSnaps
  * ConnectedSnaps inserts snapB info - it's already there so no duplication happens (avoid seting up security twice for the same snap)

The map (which is really just used a set) could just be the **```map[string]bool```** - but I thought it would be slightly more efficient to hash pointers than strings.